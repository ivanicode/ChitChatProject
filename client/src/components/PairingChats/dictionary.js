
export const interest = [
    {type: 'radio', name: 'interests', id: 1, label: 'Losowo (bez wględu na zainteresowania)'},
    {type: 'radio', name: 'interests', id: 2, label: 'Według moich zaintersowań'},
    {type: 'radio', name: 'interests', id: 3, label: 'Według mojego pierwszego zainteresowania:'},
    {type: 'radio', name: 'interests', id: 4, label: 'Według mojego drugiego zainteresowania:'},
    {type: 'radio', name: 'interests', id: 5, label: 'Według mojego trzeciego zainteresowania:'},
]

export const genderPrefference = [
    {id: 1, label: 'Losowo'},
    {id: 2, label: 'Mężczyzna'},
    {id: 3, label: 'Kobieta'},
    {id: 4, label: 'Inna'},
]

export const ageDifference = [
    {id: 1, label: 'Losowo'},
    {id: 2, label: 'Tylko w moim wieku'},
    {id: 3, label: 'Do 1 roku różnicy'},
    {id: 4, label: 'Do 2 lat różnicy'},
    {id: 5, label: 'Do 3 lat różnicy'},
    {id: 6, label: 'Do 5 lat różnicy'},
    {id: 7, label: 'Do 7 lat różnicy'},
    {id: 8, label: 'Do 10 lat różnicy'},
]
