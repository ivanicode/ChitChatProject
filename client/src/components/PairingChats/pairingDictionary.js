/*import { useFetch } from '../../common/hooks/useFetchHook'

function getRandomIntInclusive(min, max) {
  let randomNumber;
  do {
    randomNumber = Math.floor(Math.random() * (max - min + 1)) + min;
  }  while (randomNumber === getRandomIntInclusive.last)
  getRandomIntInclusive.last = randomNumber

  return randomNumber;
}

const {data} = useFetch('/api/user/details');
const {data: accountData} = useFetch('/api/user')

export const interestsPairingObject = {
  1: null,
  2: data?.interests,
  3: data?.interests.split(',')[0],
  4: data?.interests.split(',')[1],
  5: data?.interests.split(',')[2]
}

const genderPairingObject = {
  1: null,
  2: 2,
  3: 1,
  4: 3
}

const agePairingObject = {
  1: null,
  2: accountData?.birth
}*/