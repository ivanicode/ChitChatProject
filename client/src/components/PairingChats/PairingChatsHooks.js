import {useHistory} from 'react-router-dom';
import {useState, useEffect} from 'react';
import { useSave } from '../../common/hooks/useSaveHook';
import { hobby } from '../RegisterProfile/dictionary'


export function usePairingHooks(){
    const history = useHistory();
    const {
        formData,
        onChangeHandler
    } = useManageFormData()

    const {saveData} = useSave('/api/user/details2')

    const {
        submitPairingChats
    } = useSubmitPairingChats(saveData, formData, history)
    const chosenHobbys = history.location.search.substring(1).split('&').find((el) => el.includes('interest')).split('=')[1];
    const hobbys = hobby.filter((element) => chosenHobbys.includes(element.id))

    return {
        submitPairingChats,
        onChangeHandler,
        formData,
        hobbys
    }
}

export function useSubmitPairingChats(saveData, formData, history){
    
    
    function checkIfFormIsValid(){
        const valuesForValidation = Object.values(formData).find( function (value){
            return value === '';
        })
        return valuesForValidation === undefined;
    }

    const [formIsValid, setFormIsValid] = useState(checkIfFormIsValid())

    useEffect(() => {
        setFormIsValid(checkIfFormIsValid())
    }, [formData])

    function submitPairingChats(){
        if(formIsValid){
            function redirectToHome(){
                history.push('/home') 
            }
            saveData({data: formData, onSuccess: redirectToHome});
        }
    }
    return {submitPairingChats}
}

export const initialState = {
    distance: '',
    interests: '',
    gender: '',
    age: ''
}

export function useManageFormData(){

    const [formData, setFormData] = useState(initialState);

    function onChangeHandler(event){
        const id = event.target.id;
        const value = event.target.value;
        const newState = {...formData};
        newState[id] = value;
        setFormData(newState);
    }
    return {formData, onChangeHandler}
}
