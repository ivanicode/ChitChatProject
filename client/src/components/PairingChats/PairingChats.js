import React from 'react';
import {distance, genderPrefference, ageDifference, interest} from './dictionary';
import { string } from 'prop-types';
import {usePairingHooks} from './PairingChatsHooks';


export function PairingChats(){

    const {
        submitPairingChats,
        formData,
        onChangeHandler,
        hobbys
    } = usePairingHooks()

    return (
        <form className="relationship-form">
            <div className="relationshipFormDiv">
                <div className="h3Div">
                    <h3>Ostatnim krokiem do rozpoczęcia korzystania z ChitChat pozostaje ustawić parametry dla jakich będziemy znajdywać dla Ciebie ChitChaterów</h3>
                </div>
                <div className="distanceDiv">
                    <p>Odległość:</p>
                    <p>Ustaw jaka odległość powinna dzielić Cię od rozmówcy</p>
                    <select onChange={onChangeHandler} id="distance" value={formData.distance}>
                        {distance.map(el => (
                            <option value={el.id} key={el.id}>{el.label}</option>
                        ))}
                    </select>
                </div>
                <div className="interestDiv">
                    <p>Zainteresowania:</p>
                    <p>Ustaw w jaki sposób chaty mają się parować pod względem zainteresowań</p>
                        {interest.map((el, index) => (
                            <div key={el.id}>
                            <input type={el.type} name={el.name} value={el.id} onChange={onChangeHandler} id="interests"/>{el.label} {hobbys[index-2] && hobbys[index-2].label}
                            </div>
                        ))}
                </div>
                <div className="genderSelectDiv">
                    <p>Płeć:</p>
                    <p>Możesz zdecydować jakiej płci mają być Twoi rozmówcy</p>
                    <select onChange={onChangeHandler} id="gender" value={formData.gender}>
                        {genderPrefference.map(el => (
                            <option value={el.id} key={el.id}>{el.label}</option>
                        ))}
                    </select>
                </div>
                <div className="ageDifferenceDiv">
                    <p>Wiek:</p>
                    <p>Zdecyduj jaka różnica wieku ma być między Tobą a rozmówcą. Różnica będzie wynosiła maksymalnie 10lat</p>
                    <select onChange={onChangeHandler} id="age" value={formData.age}>
                        {ageDifference.map(el => (
                            <option value={el.id} key={el.id}>{el.label}</option>
                        ))}
                    </select>
                </div>
            </div>
            <div>
                <button type="button" className="submitPairingChats" onClick={submitPairingChats}>Dalej</button>
            </div>
        </form>
    )
}

PairingChats.displayName = 'PairingChats';
PairingChats.propTypes = {
    userName: string
}

export default PairingChats;