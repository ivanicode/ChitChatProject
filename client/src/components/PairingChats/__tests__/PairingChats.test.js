import React from 'react';
import {shallow} from 'enzyme';
import {PairingChats} from '../PairingChats';
import * as hooks from '../pairingChatsHooks';

describe('PairingChats component', () => {
    it('should match snapshot when no errors', () => {
        hooks.usePairingHooks = jest.fn().mockImplementation(() => ({
            submitPairingChats: jest.fn(),
            formData: {},
            onChangeHandler: jest.fn(),
            hobbys: []
        }))
        const wrapper = shallow(<PairingChats />)
        expect(wrapper).toMatchSnapshot()
    })
})