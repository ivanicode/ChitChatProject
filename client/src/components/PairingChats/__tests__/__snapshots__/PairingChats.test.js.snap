// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PairingChats component should match snapshot when no errors 1`] = `
<form
  className="relationship-form"
>
  <div
    className="relationshipFormDiv"
  >
    <div
      className="h3Div"
    >
      <h3>
        Ostatnim krokiem do rozpoczęcia korzystania z ChitChat pozostaje ustawić parametry dla jakich będziemy znajdywać dla Ciebie ChitChaterów
      </h3>
    </div>
    <div
      className="distanceDiv"
    >
      <p>
        Odległość:
      </p>
      <p>
        Ustaw jaka odległość powinna dzielić Cię od rozmówcy
      </p>
      <select
        id="distance"
        onChange={[MockFunction]}
      >
        <option
          key="1"
          value={1}
        >
          Losowo
        </option>
        <option
          key="2"
          value={2}
        >
          Tylko moja miejscowość
        </option>
        <option
          key="3"
          value={3}
        >
          do 10km
        </option>
        <option
          key="4"
          value={4}
        >
          do 20km
        </option>
        <option
          key="5"
          value={5}
        >
          do 30km
        </option>
        <option
          key="6"
          value={6}
        >
          do 40km
        </option>
        <option
          key="7"
          value={7}
        >
          do 50km
        </option>
        <option
          key="8"
          value={8}
        >
          do 60km
        </option>
        <option
          key="9"
          value={9}
        >
          do 70km
        </option>
        <option
          key="10"
          value={10}
        >
          do 80km
        </option>
        <option
          key="11"
          value={11}
        >
          do 90km
        </option>
        <option
          key="12"
          value={12}
        >
          do 100km
        </option>
        <option
          key="13"
          value={13}
        >
          do 150km
        </option>
        <option
          key="14"
          value={14}
        >
          do 200km
        </option>
        <option
          key="15"
          value={15}
        >
          do 300km
        </option>
        <option
          key="16"
          value={16}
        >
          do i powyżej 400km
        </option>
      </select>
    </div>
    <div
      className="interestDiv"
    >
      <p>
        Zainteresowania:
      </p>
      <p>
        Ustaw w jaki sposób chaty mają się parować pod względem zainteresowań
      </p>
      <div
        key="1"
      >
        <input
          id="interests"
          name="interests"
          onChange={[MockFunction]}
          type="radio"
          value={1}
        />
        Losowo (bez wględu na zainteresowania)
         
      </div>
      <div
        key="2"
      >
        <input
          id="interests"
          name="interests"
          onChange={[MockFunction]}
          type="radio"
          value={2}
        />
        Według moich zaintersowań
         
      </div>
      <div
        key="3"
      >
        <input
          id="interests"
          name="interests"
          onChange={[MockFunction]}
          type="radio"
          value={3}
        />
        Według mojego pierwszego zainteresowania:
         
      </div>
      <div
        key="4"
      >
        <input
          id="interests"
          name="interests"
          onChange={[MockFunction]}
          type="radio"
          value={4}
        />
        Według mojego drugiego zainteresowania:
         
      </div>
      <div
        key="5"
      >
        <input
          id="interests"
          name="interests"
          onChange={[MockFunction]}
          type="radio"
          value={5}
        />
        Według mojego trzeciego zainteresowania:
         
      </div>
    </div>
    <div
      className="genderSelectDiv"
    >
      <p>
        Płeć:
      </p>
      <p>
        Możesz zdecydować jakiej płci mają być Twoi rozmówcy
      </p>
      <select
        id="gender"
        onChange={[MockFunction]}
      >
        <option
          key="1"
          value={1}
        >
          Losowo
        </option>
        <option
          key="2"
          value={2}
        >
          Mężczyzna
        </option>
        <option
          key="3"
          value={3}
        >
          Kobieta
        </option>
        <option
          key="4"
          value={4}
        >
          Inna
        </option>
      </select>
    </div>
    <div
      className="ageDifferenceDiv"
    >
      <p>
        Wiek:
      </p>
      <p>
        Zdecyduj jaka różnica wieku ma być między Tobą a rozmówcą. Różnica będzie wynosiła maksymalnie 10lat
      </p>
      <select
        id="age"
        onChange={[MockFunction]}
      >
        <option
          key="1"
          value={1}
        >
          Losowo
        </option>
        <option
          key="2"
          value={2}
        >
          Tylko w moim wieku
        </option>
        <option
          key="3"
          value={3}
        >
          Do 1 roku różnicy
        </option>
        <option
          key="4"
          value={4}
        >
          Do 2 roku różnicy
        </option>
        <option
          key="5"
          value={5}
        >
          Do 3 roku różnicy
        </option>
        <option
          key="6"
          value={6}
        >
          Do 4 roku różnicy
        </option>
        <option
          key="7"
          value={7}
        >
          Do 5 roku różnicy
        </option>
        <option
          key="8"
          value={8}
        >
          Do 6 roku różnicy
        </option>
        <option
          key="9"
          value={9}
        >
          Do 7 roku różnicy
        </option>
        <option
          key="10"
          value={10}
        >
          Do 8 roku różnicy
        </option>
        <option
          key="11"
          value={11}
        >
          Do 9 roku różnicy
        </option>
        <option
          key="12"
          value={12}
        >
          Do 10 roku różnicy
        </option>
      </select>
    </div>
  </div>
  <div>
    <button
      className="submitPairingChats"
      onClick={[MockFunction]}
      type="button"
    >
      Dalej
    </button>
  </div>
</form>
`;
